### ISSUE TITLE
A model won't be updated if attribute values are all the same on save

### ISSUE THREAD

**Issue by imalitavakoli** [laravel/framework#10991]
Let's save a new user for the first time
```
$user = new User;
$user->name = 'Ali';
$user->is_block = 0;
$user->save();
```

Ok good! Now the below update won't change its 'updated_at' column which is ok(I guess), because the value of all the attributes are all the same as before... So technically we're not updating anything!
```
$user = User::find(1);
$user->name = 'Ali';
$user->is_block = 0;
$user->save();
```

But the following update **will** change its 'updated_at' column! What is the difference between these two editions? Well, this time we're using the casted `is_block` attribute instead of simply inserting 0 or 1 as a value.
```
$user = User::find(1);
$user->name = 'Ali';
$user->is_block = false;
$user->save();
```

So I guess I'm seeing a problem here!
Basically one of the following should happen in such a case:
- None of the above editions to a model should change the 'updated_at' column of that model.
- Or both of the above editions should change the 'updated_at' column.

**Comment by jairovsky** [comment]
It's because `0 !== false`.

The Model class make strict comparisons to check which properties are "dirty". ([See source code here](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Database/Eloquent/Model.php#L3144))

**Comment by imalitavakoli** [comment]
Yes, that's correct! But maybe an additional check or a patch for something like this might be a good idea...

**Comment by GrahamCampbell** [comment]
Thanks, but this is actually a duplicate.
